pkg/kubectl/cmd/delete.go

```diff
@@ -123,17 +124,6 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 	}
 
 	ignoreNotFound := cmdutil.GetFlagBool(cmd, "ignore-not-found")
-	if deleteAll {
```

Comment:
I think that is the current behavior… ignore not found by default when deleting from a calculated list, surface not found be default when deleting by name.